### PR TITLE
Fix corner case for system/staging test jobs

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review and merge.

This fixes a corner case where we compute versions to run in system and staging tests, where the first production deployment used to compute the versions has a newer platform that what we are deploying — this confused the algorithm into believing it needed both a run which was a success both on the deploying change, AND on the newer version, which is obviously impossible. When a newer version exists in the production deployment, this is now used instead of, not in addition to, the deploying change.

### CODE REVIEW

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTriggerTest.java [hunk 1]
```diff
@@ -1216,4 +1217,42 @@ public void testEagerTests() {
         app.assertNotRunning(stagingTest);
     }
 
+    @Test
+    public void test() {
```
Comment (mpolden) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTriggerTest.java:1221]: I think there's a better name to be found.

Comment (jonmv) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTriggerTest.java:1221]: 😂

Comment (jonmv) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTriggerTest.java:1221]: Had no idea what the problem was when I started writing the test ...

Comment (jonmv) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTriggerTest.java:1221]:
```suggestion
    public void testTriggeringOfIdleTestJobsWhenFirstDeploymentIsOnNewerVersionThanChange() {
```

**Review by mpolden — APPROVED** [review]

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTriggerTest.java (hunk 1 above)
Comment (mpolden) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTriggerTest.java]: Sounds familiar. 🙂
